lib/engine/game/g_1880/step/check_fi_connection.rb

```diff
@@ -96,7 +96,7 @@ def process_token(action)
           end
 
           def discard_token(fi)
-            fi.tokens.first.city.remove_tokens!
+            fi.tokens.first.remove!
```

Comment:
There is more to this one (corp needed to be closed using close_corporation). I've addressed it. Please remove this change from the PR.